File: testSuite/regressions/cosmicEmu.pl
#!/usr/bin/env perl
use strict;
use warnings;
use PDL;
use PDL::IO::HDF5;

# Run a power spectrum task using the CosmicEmu nonlinear power spectrum twice. This tests that a pre-computed power spectrum file
# is correctly re-read on subsequent runs.
# Andrew Benson (13-October-2023)

# Run the model and check for successful completion.
for(my $i=0;$i<2;++$i) {
    system("./Galacticus.exe testSuite/regressions/cosmicEmu.xml");
    die("FAILED: cosmicEmu.pl model failed to complete") 
	unless ( $? == 0 );
}

exit;

File: testSuite/regressions/cosmicEmu.py
#!/usr/bin/env python3
import subprocess
import sys

# Run a power spectrum task using the CosmicEmu nonlinear power spectrum twice. This tests that a
# pre-computed power spectrum file is correctly re-read on subsequent runs.
# Andrew Benson (13-October-2023; ported to Python)

# Run the model and check for successful completion.
for i in range(2):
    status = subprocess.run("./Galacticus.exe testSuite/regressions/cosmicEmu.xml", shell=True)
    if status.returncode != 0:
        print("FAILED: cosmicEmu model failed to complete")
        sys.exit(1)

print("SUCCESS: cosmicEmu")
